src/Adapter/Product/AdminProductWrapper.php

```diff
@@ -706,16 +705,33 @@ public function getPreviewUrl($product)
             $is_rewrite_active
         );
 
-        if (!$product->active) {
-            $admin_dir = dirname($_SERVER['PHP_SELF']);
-            $admin_dir = substr($admin_dir, strrpos($admin_dir, '/') + 1);
-            $preview_url .= ((strpos($preview_url, '?') === false) ? '?' : '&').'adtoken='.$token.'&ad='.$admin_dir.'&id_employee='.(int)$context->employee->id;
+        if (!$product->active && $preview) {
+            $preview_url = $this->getPreviewUrlDeactivate($preview_url);
         }
 
         return $preview_url;
     }
 
     /**
+     * Generate preview URL deactivate
+     *
+     * @param string $preview_url
+     *
+     * @return string preview url deactivate
+     */
+    public function getPreviewUrlDeactivate($preview_url)
+    {
+        $context = \ContextCore::getContext();
+        $token = \ToolsCore::getAdminTokenLite('AdminProducts');
+
+        $admin_dir = dirname($_SERVER['PHP_SELF']);
```

Comment:
uhuh ^^ what do you expect ? did you take a look at the available constants in `defines.inc.php` file ?